typing/typecore.ml

```diff
@@ -526,9 +537,9 @@ let enter_orpat_variables loc env  p1_vs p2_vs =
           (x2,x1)::unify_vars rem1 rem2
           end
       | [],[] -> []
-      | (x,_,_,_,_)::_, [] -> raise (Error (loc, env, Orpat_vars (x, [])))
-      | [],(y,_,_,_,_)::_  -> raise (Error (loc, env, Orpat_vars (y, [])))
-      | (x,_,_,_,_)::_, (y,_,_,_,_)::_ ->
+      | {pv_id = x; _}::_, [] -> raise (Error (loc, env, Orpat_vars (x, [])))
+      | [],{pv_id = y; _}::_  -> raise (Error (loc, env, Orpat_vars (y, [])))
```

Comment:
While you're touching these two lines (and I'm making stylistic comments), do you mind merging the two cases above?